### PR TITLE
Add DeepL as translation service

### PR DESCRIPTION
**Pull request by Elhebert**
Hi 👋 

Nice package! 

I figured it would be nice to also include DeepL as a translation service. This will give an opportunity to the privacy focused people an alternative to OpenAI and Google Translate.

I’m marking the PR as a draft since I want to test the integration in a real Laravel app before I mark the PR as done.

### CODE REVIEW

File: tests/Translator/DeeplTranslatorTest.php [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    public function testDeeplTranslatorBinding(): void
+    {
+        $this->assertInstanceOf(DeeplTranslator::class, app(DeeplTranslator::class));
+    }
+
+    #[Test]
+    public function testDeeplTranslatorHasValidCredentials(): void
+    {
+        $this->assertTrue($this->deeplTranslator->isConfigured());
+        ;
```
Comment (Bottelet) [inline tests/Translator/DeeplTranslatorTest.php:94]: A semi colon too much :)

File: tests/Translator/DeeplTranslatorTest.php [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    }
+
+    #[Test]
+    public function testDeeplTranslatorHasInvalidCredentials(): void
+    {
+        $this->app['config']->set('translator.translators.deepl', [
+            'api_key' => null,
+        ]);
+        $this->assertFalse($this->deeplTranslator->isConfigured());
+        ;
```
Comment (Bottelet) [inline tests/Translator/DeeplTranslatorTest.php:104]: semi colon

**Comment by Bottelet** [comment]
This is great! Let me know how the testing goes, and i will merge it.

File: tests/Translator/DeeplTranslatorTest.php (hunk 1 above)
Comment (Elhebert) [reply tests/Translator/DeeplTranslatorTest.php:94]: I copied this from your existing tests. I thought it was odd but didn’t want to deviate from your style. I’ll remove them 👍

**Comment by Elhebert** [comment]
@Bottelet I did a test locally on a personal project, works like a charm with DeepL 👌

**Comment by Bottelet** [comment]
This is great I appreciate the contribution! :raised_hands:

**Pull request merged**